docs/cli: Added test command docs.

suitupalex opened the pull request:
The documentation for the `test` command was missing so I decided to add it in with the first commit.

~~In the second commit, I added more information about system-wide commands not being executable by Yarn. Since it's possibly a breaking change for people coming from npm, I felt it should be proactively addressed in the docs.~~

~~The third commit just adds a missing `p.lead` tag to `run`.~~

I tried to lift the wording and formatting from other doc entries.

cpojer commented:
Would you mind reverting the changes to `yarn run`? That seems to me like a bug we should fix in yarn. Let's keep this PR focused on `yarn test`.

addyosmani commented:
> Let's keep this PR focused on yarn test.

👍

suitupalex commented:
Removed both the changes to `run` docs and mention of the inability to run system commands. Looks like that issue was found and fixed in yarnpkg/yarn#733.

cpojer commented:
thank you!
